### PR TITLE
[CP Staging] Fix issues with scrollbar and status bar color

### PR DESCRIPTION
**Pull request by chrispader**
@grgia 

### Details

Fixes multiple issues with status bar, color scheme and web background after theme switching, e.g. emoji/mention selection on Windows web.

### Fixed Issues

$ https://github.com/Expensify/App/issues/32591
$ https://github.com/Expensify/App/issues/32695
$ https://github.com/Expensify/App/issues/32657
PROPOSAL: 

### Tests

- [ ] Verify that no errors appear in the JS console

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Comment by grgia** [comment]
@chrispader can we revert these as part of this PR and address any native issues with the wrapper as well? 
https://github.com/Expensify/App/pull/13874/files#diff-bc1549da6355f7f82f324af9c667b78ff60672b6b80219062629fe4184e8fa93R69-R70

**Comment by grgia** [comment]
@situchan since you've been on top of these issues, mind I add you as C+?

**Comment by situchan** [comment]
> @situchan since you've been on top of these issues, mind I add you as C+?

sure thanks. will review once ready for review

**Comment by grgia** [comment]
@parasharrajat shall I add you as C+ as well?

**Comment by grgia** [comment]
These minor issues are all related to the same root cause for tracking:
https://github.com/Expensify/App/issues/32695
https://github.com/Expensify/App/issues/32657
https://github.com/Expensify/App/pull/32335#issuecomment-1839009952

**Comment by situchan** [comment]
Can you please complete checklist, link fixed issues correctly and fix lint?

**Comment by chrispader** [comment]
> Can you please complete checklist, link fixed issues correctly and fix lint?

done 👍

**Comment by situchan** [comment]
reviewing

File: src/Expensify.js [hunk 1]
```diff
@@ -202,6 +204,11 @@ function Expensify(props) {
         // eslint-disable-next-line react-hooks/exhaustive-deps -- we don't want this effect to run again
     }, []);
 
+    useEffect(() => {
+        const htmlElement = document.getElementsByTagName('html')[0];
```
[the remark this answers is not in the document]
Comment (chrispader) [reply src/Expensify.js:208]: ah whoopsie, thanks!

**Comment by situchan** [comment]
https://github.com/Expensify/App/issues/32657 requires native code changes. Is it supposted to fix here?

File: src/Expensify.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrispader) [reply src/Expensify.js]: updated 👍

**Comment by chrispader** [comment]
> #32657 requires native code changes. Is it supposted to fix here?

wdym by it requires native code changes? The issue in https://github.com/Expensify/App/issues/32657 is just that the `CustomStatusBar` component wasn't updated, when the theme changes, but only on navigation change. Fixed that in this PR

**Comment by situchan** [comment]
Bug1: [iOS] status bar glitch after changing theme


https://github.com/Expensify/App/assets/108292595/48480c58-e1a3-4e08-ba28-93e39fa13666

**Comment by situchan** [comment]
Bug2: [mSafari] sometimes wrong status bar color while switching LHN <-> report


https://github.com/Expensify/App/assets/108292595/54461cf6-6c15-4fa0-ad85-72108be9dee0

**Comment by situchan** [comment]
Bug3: [native] scroll bar indicator color is white in light theme. Should be dark


https://github.com/Expensify/App/assets/108292595/ef5fbb70-2ba5-409c-be70-dbf92583d7c6

**Comment by situchan** [comment]
> > #32657 requires native code changes. Is it supposted to fix here?
> 
> wdym by it requires native code changes? The issue in #32657 is just that the `CustomStatusBar` component wasn't updated, when the theme changes, but only on navigation change. Fixed that in this PR

Hmm, that's fine if status bar theme can be fully customized in js. But I don't think so for native theme changes like scroll bar indicator.

**Comment by chrispader** [comment]
> Bug3: [native] scroll bar indicator color is white in light theme. Should be dark
> 
>  Screen.Recording.2023-12-14.at.2.28.14.AM.mov

This one we'll have to fix in a follow-up PR as it (most probably) involves native code changes.

**Comment by chrispader** [comment]
I fixed bugs 1 and 2 :)

@situchan please review again

**Comment by grgia** [comment]
I created an issue for the IOS issue discussed: https://github.com/Expensify/App/issues/33037

**Comment by situchan** [comment]
Native part in https://github.com/Expensify/App/issues/32591 should be handled in https://github.com/Expensify/App/issues/33037 as well. Actually same issue.


https://github.com/Expensify/App/assets/108292595/f8df97cb-906a-4da4-bb1c-dd340d66bceb

**Comment by situchan** [comment]
Please add all the cases fixed here in QA Steps

**Comment by situchan** [comment]
Please pull main as #21669 is merged

**Comment by situchan** [comment]
Not sure if this should be fixed here or out of scope, but some parts are not applied real time when change system theme
(tested after pull main which applied https://github.com/Expensify/App/pull/21669)

Otherwise looks good!


https://github.com/Expensify/App/assets/108292595/f1bdf002-fbb8-41ac-bb9d-98a343fd3a79

**Comment by situchan** [comment]
This maybe also related to native theme change in https://github.com/Expensify/App/issues/33037. Btw out of scope

Keyboard theme should stick to app theme, not device theme


https://github.com/Expensify/App/assets/108292595/2950a76a-2c89-4c91-86c1-e23b28ff31d0

**Comment by situchan** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


https://github.com/Expensify/App/assets/108292595/6a3bd483-35b4-4005-9366-db12ab289a93



</details>

<details>
<summary>Android: mWeb Chrome</summary>


https://github.com/Expensify/App/assets/108292595/72fa84d1-877b-45e4-8a45-24320db0b028



</details>

<details>
<summary>iOS: Native</summary>


https://github.com/Expensify/App/assets/108292595/afdefd37-df3c-4a8b-a88e-f3678120fbf2



</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/108292595/74241d17-1d86-4d57-92cc-63d9f012a9bf




</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/Expensify/App/assets/108292595/f25389a8-f8c8-4279-995a-617675bd66cb



</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/108292595/0fb78997-a35c-40c7-b52d-c5bd8f4c2979


https://github.com/Expensify/App/assets/108292595/2a250479-a34a-43a6-badb-96f17a254202




</details>

**Comment by chrispader** [comment]
> Not sure if this should be fixed here or out of scope, but some parts are not applied real time when change system theme (tested after pull main which applied #21669)

yea, we can work on this in a follow-up PR

**Comment by chrispader** [comment]
> This maybe also related to native theme change in #33037. Btw out of scope
> 
> Keyboard theme should stick to app theme, not device theme

I can only set this on IOS through the `keyboardAppearance` prop on `TextInput`.

Afaik, there is no official way to set the keyboard theme on Android, since Android allows "soft keyboards" and there technically is no protocol or system keyboard, just custom or preinstalled ones.

https://stackoverflow.com/a/39705139/5774312

We might be able to somehow natively change the keyboard theme, but developers of custom keyboards might still ignore it, so idk if we really want to pursue this

**Review by situchan — APPROVED** [review]

**Review by grgia — APPROVED** [review]

**Comment by grgia** [comment]
> Not sure if this should be fixed here or out of scope, but some parts are not applied real time when change system theme (tested after pull main which applied #21669)
> 
> Otherwise looks good!
> 
>  Screen.Recording.2023-12-14.at.10.16.43.AM.mov

Issue created: https://github.com/Expensify/App/issues/33071

**Comment by grgia** [comment]
> Keyboard theme should stick to app theme, not device theme

> Afaik, there is no official way to set the keyboard theme on Android, since Android allows "soft keyboards" and there technically is no protocol or system keyboard, just custom or preinstalled ones.



@situchan @chrispader I wonder if other react native apps with dark/light mode handle this (i.e discord) 
I agree it's polish and I think it looks okay if the keyboard matches the user's native OS- though if we can fix this, it would be nice polish. 

Is it worth creating an issue for this?

**Comment by grgia** [comment]
@chrispader before I merge, could you add QA steps?

**Comment by situchan** [comment]
> @situchan @chrispader I wonder if other react native apps with dark/light mode handle this (i.e discord)
I agree it's polish and I think it looks okay if the keyboard matches the user's native OS- though if we can fix this, it would be nice polish.
>
> Is it worth creating an issue for this?

Not worth

**Comment by jjcoffee** [comment]
Coming from https://github.com/Expensify/App/issues/33674, where the status bar style wasn't correctly updating on initial app launch. This is due to setting the initial state of the status bar here:
https://github.com/Expensify/App/blob/376bb9563618502906032a970b09c0075c40a5d5/src/components/CustomStatusBarAndBackground/index.tsx#L18

But the status bar color isn't updated via `updateStatusBarAppearance` yet, so that initial state causes the status bar to not to change color due to [this logic](https://github.com/Expensify/App/blob/376bb9563618502906032a970b09c0075c40a5d5/src/components/CustomStatusBarAndBackground/index.tsx#L60). More details [here](https://github.com/Expensify/App/issues/33674#issuecomment-1870717852).

**Pull request merged**